### DISCUSSION TITLE
Topics to work on for the sprint

### DISCUSSION THREAD

**Discussion by The-Compiler** [pytest-dev/sprint#3] [General]
Other than people who can mentor (#2), obviously we'll also need stuff to work on!

Are there topics you'd like to see at the sprint, or you feel like would be nice to tackle?

From the top of my head:

- https://github.com/pytest-dev/pytest/issues/3834 is something I'd be interested in. Can we maybe have class-based fixtures of some sorts in a backwards-compatible way?
- https://github.com/pytest-dev/pytest/issues/10032 (assert helpers) is another one
- https://github.com/pytest-dev/pytest/issues/11538 (matchers for pytest.raises) is one that came up in the past, and might be interesting for @RonnyPfannschmidt and @asottile to collaborate on (assuming both are still coming to the sprint)?

**Comment by RonnyPfannschmidt** [comment]
Topics I'm also interested in

- collection roots 
- traceback filtering/culling
- integration of function definition 
- non python support for generate-tests

I fear that non name based fixtures may be a bit too big but I'm very enthusiastic about creating a better base for fixture definitions and lookup to make lookup by type/name as well as explicit reference more easy

**Comment by okken** [comment]
Something I've never talked about with anyone, but am curious if it's possible. ...
`pytest --help` is huge, and needs a bit of organizing, I think.

For example, at the bottom is 
```
to see available markers type: pytest --markers
to see available fixtures type: pytest --fixtures
```

I think it'd be cool to have more stuff like that. 

Perhaps, by default, we can hide away non-newbie stuff, like:
- all of the ini file settings
- pastebin (does anyone use that?)
- junit stuff
- doctest flags
- log flags
- advanced features (confcutdir, import-mode, keep-duplicates, ...)
- deprecated features 

Then perhaps we could make sure the rest of them are correctly grouped. 
- Seems like --sw, --sw-skip, --trace, --pdb maybe belong in a "test debugging" group?
- Wouldn't -k and -m belong in collection?
- Do we need a "general" category?
- I'm sure there are others

I think making sure the default `pytest --help` is fairly beginner friendly would make it easier for people to get started.



-

**Reply by hugovk** [reply]
Would be nice to see some ANSI bold or colour in `--help`, respecting the current env vars and options to disable/force colour. Compare `ruff --help` or `uv --help`.

**Reply by webknjaz** [reply]
@hugovk FYI the `FORCE_COLOR` and `NO_COLOR` standard authors can't decide what to do with the corner cases: https://github.com/donatj/force-color.org/issues/8. At least the `NO_COLOR` author seems to think that **bold** is not a color: https://github.com/Textualize/rich/pull/2449#issuecomment-1820994887.
